hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -546,12 +568,70 @@ public List<WriteStatus> close() {
         status.getStat().setFileSizeInBytes(logFileSize);
       }
 
+      // generate Secondary index stats if streaming is enabled.
+      if (generateStatsForStreamingMetadataWrites) {
+        trackMetadataIndexStatsForStreamingMetadataWrites(fileSliceOpt.or(getFileSlice()), statuses.stream().map(status -> status.getStat().getPath()).collect(Collectors.toList()),
+            statuses.get(statuses.size() - 1));
```

Comment:
can you add java docs on why we update just the last WriteStatus when we have more than 1 WriteStatus

Reply:
Discussed offline. Updated the javadocs